komorebi/wallpaper_creator/window.py

```diff
@@ -0,0 +1,214 @@
+		self.yes_btn = Gtk.Button(label='Yes')
+
+		self.revealer = Gtk.Revealer()
+		self.info_bar = Gtk.InfoBar(message_type=Gtk.MessageType.ERROR)
+		self.error_lbl = Gtk.Label()
+
+		# Setup window
+		self.set_size_request(1050, 700)
+		self.set_resizable(False)
+		self.set_position(Gtk.WindowPosition.CENTER)
+		self.set_titlebar(self.header_bar)
+		apply_css([self.layout], self.main_css)
+		apply_css([self.header_bar], self.header_css)
+		apply_alpha([self])
+
+		# Setup widgets
+		self.close_btn.set_margin_top(6)
+		self.close_btn.set_margin_start(6)
+		self.close_btn.set_halign(Gtk.Align.START)
+
+		self.add_layer_btn.set_margin_top(6)
+		self.add_layer_btn.set_margin_start(6)
+		self.add_layer_btn.set_halign(Gtk.Align.START)
+
+		self.next_btn.set_margin_top(6)
+		self.next_btn.set_margin_end(6)
+
+		self.popover.set_relative_to(self.close_btn)
+
+		self.popover_grid.set_margin_top(15)
+		self.popover_grid.set_margin_bottom(15)
+		self.popover_grid.set_margin_left(15)
+		self.popover_grid.set_margin_right(15)
+		self.popover_grid.set_row_spacing(20)
+		self.popover_grid.set_column_spacing(5)
+
+		self.revealer.set_transition_duration(200)
+		self.revealer.set_transition_type(Gtk.RevealerTransitionType.SLIDE_DOWN)
+
+		self.stack.set_transition_duration(400)
+		self.stack.set_transition_type(Gtk.StackTransitionType.SLIDE_LEFT)
+
+		# Setup signals
+		def _on_close_btn_released(_, self):
+			self.popover.show_all()
+
+		def _on_add_layer_btn_released(_, self):
+			file_choose_dialog = Gtk.FileChooserDialog(title='Select an image', parent=self, action=Gtk.FileChooserAction.OPEN)
+			file_choose_dialog.add_buttons(
+				'Cancel', Gtk.ResponseType.CANCEL,
+				'Accept', Gtk.ResponseType.ACCEPT
+			)
+
+			file_filter = Gtk.FileFilter()
+			file_filter.add_mime_type('image/*')
+			file_choose_dialog.set_filter(file_filter)
+
+			if file_choose_dialog.run() == Gtk.ResponseType.ACCEPT:
+				utilities.asset_path = file_choose_dialog.get_file().get_path()
+				self.options_page.set_asset(utilities.asset_path)
+			
+			file_choose_dialog.close()
+		
+		def _on_next_btn_released(_, self):
+			current_page = self.stack.get_visible_child_name()
+
+			if current_page == 'initial':
+				if utilities.wallpaper_name == None or \
+					(utilities.wallpaper_type == 'image' or utilities.wallpaper_type == 'video') and \
+					utilities.file_path == None:
+					self.display_error('Please enter a wallpaper name and choose a file')
+					return True
+				elif utilities.wallpaper_name == None or utilities.wallpaper_type == 'web_page' and \
+					utilities.web_page_url == None:
+					self.display_error('Please enter a wallpaper name, a valid URL, and a thumbnail')
+					return True
+				
+				self.options_page = pages.OptionsPage()
+				self.add_layer_btn.set_visible(utilities.wallpaper_type == 'image')
+				self.options_page.set_image(utilities.file_path) if utilities.wallpaper_type == 'image' else self.options_page.set_blank()
```

Comment:
Since you are just running code, a normal if block would make this easier to read.

Reply:
Fixed

Reply:
Thanks!